integration/container_stats_test.go

```diff
@@ -69,6 +69,44 @@ func TestContainerStats(t *testing.T) {
 	testStats(t, s, containerConfig)
 }
 
+func TestContainerStatsDoesNotErrorOnStoppingContainer(t *testing.T) {
+	t.Logf("Create a pod config and run sandbox container")
+	sb, sbConfig := PodSandboxConfigWithCleanup(t, "sandbox1", "stats")
+
+	EnsureImageExists(t, pauseImage)
+
+	t.Logf("Create a container config and run container in a pod")
+	containerConfig := ContainerConfig(
+		"container1",
+		pauseImage,
+		WithTestLabels(),
+		WithTestAnnotations(),
+	)
+	cn, err := runtimeService.CreateContainer(sb, containerConfig, sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.RemoveContainer(cn))
+	}()
+	require.NoError(t, runtimeService.StartContainer(cn))
+	defer func() {
```

Comment:
stopping twice?

Reply:
Overly paranoid I suppose. Will remove the cleanup function.